### PR TITLE
CHAIN-476: price feed and leaderboard table

### PR DESCRIPTION
**Pull request by bholzman**

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/apps/api/TestnetChallengeRoutes.kt [hunk 1]
```diff
@@ -128,7 +151,86 @@ class TestnetChallengeRoutes(blockchainClients: Collection<BlockchainClient>) {
...
+                        TestnetChallengePNLTable.currentBalance,
+                        UserTable.nickName,
+                        UserTable.avatarUrl,
+                        WalletTable.address,
+                        pnlRatioExpr,
+                    )
+                        .where { TestnetChallengePNLTable.type eq testnetChallengePNLType }
+                        .orderBy(pnlRatioExpr, SortOrder.DESC)
+                        .limit(rowsPerPage, offset = page * rowsPerPage)
+                        .toList()
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/apps/api/TestnetChallengeRoutes.kt:216]: Nit: we usually have this kind of code in the entity/table file. Please consider moving it.

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/TestnetChallengePNL.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    it[TestnetChallengePNLTable.type] = type
+                }
+            }
+        }
+
+        fun updateAllBalances() {
+            val referenceSymbol = TestnetChallengeUtils.depositSymbol()
+            TransactionManager.current().exec(
+                """
+                WITH balance_summary AS (
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/TestnetChallengePNL.kt:70]: Nice!

**Review by antonyshchenko — APPROVED** [review]
Looks great!

File: backend/src/main/kotlin/xyz/funkybit/apps/api/TestnetChallengeRoutes.kt (hunk 1 above)
Comment (bholzman) [reply backend/src/main/kotlin/xyz/funkybit/apps/api/TestnetChallengeRoutes.kt:216]: ah, sure, will move it

**Pull request merged**